src/fortran/urqmd-3.4/chromo_urqmd.f

```diff
@@ -62,11 +62,9 @@ SUBROUTINE CHEPEVT
 
 c UrQMD doesn't know K0S or L and produces K0(bar) by default.
 c We will replace those with K0S/L based on a 50/50 rule.
-c Random numbers are sourced from second sequence
-c to avoid interference with the event generator's sequence
 
          IF (ABS(IDHEP(I)).EQ.311) THEN
-            IDHEP(I) = K0SEL(INT(2.D0*SIMRND2()))
+            IDHEP(I) = K0SEL(INT(2.D0*SIMRND()))
```

Comment:
What about the interference with the event generator's sequence?

Reply:
What kind of interference do you worry about? I can see no reason to use a different stream here.

Reply:
This and previous question boils down to the question: if we take random numbers from one sequence and make from them 2 sequencies (like 1st number goes to 1st sequence and the next one goes to 2nd, and 3rd number goes to 1st sequence again and so on), are two sequencies still good random sequencies (have the same properties as the original)? Most probably yes, but I am not sure 100%.

Reply:
A mixed stream of pseudo-random numbers from two PRNGs with different seeds is in theory more "random" than a single stream. A true random number sequences has no autocorrelation whatsoever. Pseudo-random number sequences have autocorrelations, but they so designed so that these auto-correlations are negligible. If  a mixed stream as described in the first sentences has less autocorrelation, so it is closer to the a true randon number sequence. We don't have to worry about this, our PRNG is so good that we do not need to mix numbers from several streams.

In this particular line that we are discussing here, the intend (as explained in the comment) was to draw a random number from a different stream of pseudo-random numbers than the one used by URQMD, so that there is no interference. But there cannot be any interference. URQMD cannot expect the PRNG to be in any particular state, it wouldn't make sense. As a programmer point of view, you need to treat the pseudo-random numbers like a stream of true random numbers. In reality, PRNG give you some added advantages over true randomness, like the ability to make reproducible sequences of random numbers or to rewind the PRNG to an earlier state. There is no interface in URQMD to rewind the PRNG, though, so we don't have to worry about interference.

Reply:
To leave one more comment on this multiple sequences topic... The reason for choosing multiple sequences is simple. Often, the generators don't run in a vacuum and are linked to some other user code. If the user code needs random numbers (like here), it is better to use a separate sequence. If a crash occurs in event 1023 then one can pass the code to the developers of the generator and ask to have a look at event 1023. These days, we probably won't work like that and rather ask for the user code (chromo in this case). So as long there is a way to reproduce certain events or problems somehow, it is fine.

Reply:
Thanks for the enlightening comments. If I had anything in mind when explaining my concerns, it was autocorrelation. Now I realized that the interference here is just a way to separate 2 streams of random numbers.